### ISSUE TITLE
Audience Claim is an array - causing rejection from Firebase

### ISSUE THREAD

**Issue by yvettecook** [vapor/jwt#68]
This may or may not be a general issue, but raising it for visibility.

When authenticating with Firebase, you use a JWT generated on Vapor, passed to the mobile app, and then passed to Firebase. Documentation of the process and JWT specification can be found [here](https://firebase.google.com/docs/auth/admin/create-custom-tokens).

### Problem:
JWT Payload:
```json
{
  "iss": "firebase-adminsdk-x1xx1@xxxxxxxx.iam.gserviceaccount.com",
  "aud": [
    "https://identitytoolkit.googleapis.com/google.identity.identitytoolkit.v1.IdentityToolkit"
  ],
  "iat": 1499885888,
  "exp": 1499888888,
  "sub": "firebase-adminsdk-x1xx1@xxxxxxxx.iam.gserviceaccount.com",
  "uid": "xxxxxxxx"
}
```

When authenticating with Firebase from my iOS app with this JWT payload, I was receiving this error:

`ERROR: Optional(Error Domain=FIRAuthErrorDomain Code=17000 "The custom token format is incorrect. Please check the documentation." UserInfo={NSLocalizedDescription=The custom token format is incorrect. Please check the documentation., error_name=ERROR_INVALID_CUSTOM_TOKEN})`

The incorrect format was due to the `aud` field being an array, not a single string.

### Solution:

I solved this by creating a custom claim:

```swift
struct SingleAudienceClaim: EqualityClaim {
    public static var name = "aud"
    public let value: String

    init(string: String) {
        self.value = string
    }

    public init?(_ node: Node) {
        guard let string = node.string else {
            return nil
        }

        self.init(string: string)
    }

    public var node: Node {
        return .string(value)
    }
}
```

I'm new to using JWT (and server-side in general), so not certain if this simply is a very specific issue with Firebase, or may reoccur across other sources.

Another solution could be the make the naming of `AudienceClaim` more explicit that it is for multiple audience values?

**Comment by vzsg** [comment]
[RFC 7519](https://tools.ietf.org/html/rfc7519#section-4.1.3) states the following:

> In the general case, the "aud" value is an array of case-sensitive strings, each containing a StringOrURI value.  In the special case when the JWT has one audience, the "aud" value MAY be a single case-sensitive string containing a StringOrURI value.

So I think the library should indeed support both scenarios.

**Comment by tanner0101** [comment]
Would probably make sense to include `SingleAudienceClaim` and rename `AudienceClaim` to `MultipleAudienceClaim`, deprecating the name `AudienceClaim`.

**Comment by tanner0101** [comment]
Check out https://github.com/vapor/jwt/pull/71

**Comment by tanner0101** [comment]
Closed #71 due to little interest. This will not be a restriction of the JWT 3.0 API.
